libs/apps/python.go

```diff
@@ -0,0 +1,127 @@
+package apps
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const DEBUG_PORT = "5678"
+
+var defaultLibraries = []string{
+	"Flask==3.0.3",
+	"streamlit==1.38.0",
+	"gradio==4.44.0",
+	"uvicorn[standard]==0.30.6",
+	"databricks-sdk==0.33.0",
+	"databricks-sql-connector==3.7.3",
+	"debugpy",
+}
+
+type PythonApp struct {
+	config *Config
+	spec   *AppSpec
+	uvArgs []string
+}
+
+func NewPythonApp(config *Config, spec *AppSpec) *PythonApp {
+	return &PythonApp{config: config, spec: spec}
+}
+
+// PrepareEnvironment prepares the environment for running the app. It first checks if the app is in a virtual environment.
+// If not, it creates a virtual environment and installs the required libraries. It then installs the libraries from
+// requirements.txt if it exists.
+func (p *PythonApp) PrepareEnvironment() error {
+	// Create venv first
+	venvArgs := []string{"uv", "venv"}
```

Comment:
Does this take care of installing from uv native representations of dependencies as well? Like uv.lock or pyproject.toml. Do we want to support those here?

Reply:
No, it does not not, same as app runtime doesn't